api/src/main/java/org/openmrs/module/reporting/report/processor/HttpReportProcessor.java

```diff
@@ -0,0 +1,64 @@
+package org.openmrs.module.reporting.report.processor;
+
+import java.io.OutputStreamWriter;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.module.reporting.report.Report;
+import org.springframework.stereotype.Component;
+
+@Component
+public class HttpReportProcessor implements ReportProcessor {
+
+	protected Log log = LogFactory.getLog(this.getClass());
+
+	/**
+	 * @see ReportProcessor#getConfigurationPropertyNames()
+	 */
+	@Override
+	public List<String> getConfigurationPropertyNames() {
+		List<String> ret = new ArrayList<String>();
+		ret.add("connection url");
+		ret.add("subject");
+		ret.add("Add report");
+		return ret;
+	}
+
+	/**
+	 * Performs some action on the given report
+	 * 
+	 * @param report the Report to process
+	 */
+	@Override
+	public void process(Report report, Properties configuration) {
+		try {
+			URL url = new URL(configuration.getProperty("connection url"));
+			HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+			connection.setRequestMethod("POST");
+			connection.setRequestProperty("Content-Type", "text/html; charset=UTF-8");
+			connection.setDoOutput(true);
+			connection.connect();
+			 
+			if (report.getRenderedOutput() != null && "true".equalsIgnoreCase(configuration.getProperty("Add report"))) {	
+				String addcontent = configuration.getProperty(("Add report") + configuration.getProperty("subject"));
```

Comment:
I don't really understand what all of this is doing with the "Add report" and "subject" properties.  Is this meant to be like saying "enabled=true/false"?

There is no javadoc, no examples, and no unit tests showing what is expected here.  All of that needs to be added.